controllers/hetznercluster_controller_test.go

```diff
@@ -840,3 +842,100 @@ var _ = Describe("reconcileRateLimit", func() {
+	t.Run("function should return false if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled is not enabled and hetznerCluster.Spec.ControlPlaneEndpoint is nil", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: false}, ControlPlaneEndpoint: nil}}
+		got := SetControlPlaneEndpoint(hetznerCluster)
+		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled == true){
+			t.Fatalf("Wrong value for ControlPlaneLoadBalancer.Enabled. Got: %v, Want: false", hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled)
+		}
+		if (hetznerCluster.Spec.ControlPlaneEndpoint != nil){
+			t.Fatalf("Wrong value for hetznerCluster.Spec.ControlPlaneEndpoint. Got: %s, Want: <nil>", hetznerCluster.Spec.ControlPlaneEndpoint)
+		}
+		assert.Equal(t, false, got)
+	})
+	t.Run("function should return true if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled is false but hetznerCluster.Spec.ControlPlaneEndpoint is not nil", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: false}, ControlPlaneEndpoint: &clusterv1.APIEndpoint{Host: "xyz"}}}
+		got := SetControlPlaneEndpoint(hetznerCluster)
+		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled != false){
+			t.Fatalf("Wrong value for ControlPlaneLoadBalancer.Enabled. Got: %v, Want: false", hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled)
+		}
+		if (hetznerCluster.Spec.ControlPlaneEndpoint == nil){
+			t.Fatalf("ControlPlaneEndpoint must not be nil")
+		}
+		if (hetznerCluster.Spec.ControlPlaneEndpoint.Host != "xyz"){
+			t.Fatalf("Wrong input for host. Got: %s, Want: 'xyz'", hetznerCluster.Spec.ControlPlaneEndpoint.Host)
+		}
+		assert.Equal(t, true, got)
+	})
+	t.Run("function should return false if ControlPlaneLoadBalancer is enabled and hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 is '<nil>'", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: true}}, Status: infrav1.HetznerClusterStatus{ControlPlaneLoadBalancer: &infrav1.LoadBalancerStatus{IPv4: "<nil>"}}}
+		got := SetControlPlaneEndpoint(hetznerCluster)
+		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled != true){
+			t.Fatalf("Wrong value for ControlPlaneLoadBalancer.Enabled. Got: %v, Want: true", hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled)
+		}
+		if (hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 != "<nil>"){
+			t.Fatalf("Wrong value for Status.ControlPlaneLoadBalancer.IPv4. Got: %s, Want: '<nil>'", hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4)
+		}
+		assert.Equal(t, false, got)
+	})
+	t.Run("function should return true if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled is true and hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 contains a value (not nil)", func(t *testing.T) {
```

Comment:
why don't you test the changes to ControlPlaneEndpoint in this test as well?

Reply:
I think I should just remove this test. The test below checks values of controlplaneEndpoint and is pretty much similar

Reply:
then do that